backend/canisters/local_user_index/impl/src/lifecycle/post_upgrade.rs

```diff
@@ -0,0 +1,56 @@
+use std::time::Duration;
+
+use crate::lifecycle::{init_logger, init_state, UPGRADE_BUFFER_SIZE};
+use crate::{Data, LOG_MESSAGES};
+use canister_logger::{LogMessage, LogMessagesWrapper};
+use canister_tracing_macros::trace;
+use ic_cdk_macros::post_upgrade;
+use local_user_index_canister::post_upgrade::Args;
+use stable_memory::deserialize_from_stable_memory;
+use tracing::info;
+use utils::consts::MIN_CYCLES_BALANCE;
+use utils::env::canister::CanisterEnv;
+use utils::time::MINUTE_IN_MS;
+
+#[post_upgrade]
+#[trace]
+fn post_upgrade(args: Args) {
+    ic_cdk::setup();
+
+    let env = Box::new(CanisterEnv::new());
+
+    let (data, log_messages, trace_messages, cycles_dispenser_client_state): (Data, Vec<LogMessage>, Vec<LogMessage>, Vec<u8>) =
+        deserialize_from_stable_memory(UPGRADE_BUFFER_SIZE).unwrap();
+
+    let user_index_canister_id = data.user_index_canister_id;
+
+    init_logger(data.test_mode);
+    init_state(env, data, args.wasm_version);
+
+    if !log_messages.is_empty() || !trace_messages.is_empty() {
+        LOG_MESSAGES.with(|l| rehydrate_log_messages(log_messages, trace_messages, &l.borrow()))
+    }
+
+    cycles_dispenser_client::init_from_bytes(&cycles_dispenser_client_state);
```

Comment:
I think the 'local' canisters should request cycles from their parent rather than directly from the cycles dispenser.

Reply:
Yes that was a mistake